src/liboslexec/llvm_gen.cpp

```diff
@@ -360,23 +361,27 @@ LLVMGEN (llvm_gen_printf)
 
                     llvm::Value* loaded = nullptr;
                     if (rop.use_optix() && simpletype.basetype == TypeDesc::STRING) {
-                        // TODO: make this work with arrind
-                        // In the OptiX case, we use the device-side string.
-                        loaded = rop.llvm_load_device_string (sym);
+                        // In the OptiX case, we register each string separately.
```

Comment:
I'm not sure I understand how or why this section works. What's up with the "%s[%d]" part? You're making a string compiler symbol that has the "arrayname[index]" and value is the contents of that element of the string array, so you can then ask for the device string? I think I get that, it could use an explanatory comment. Also, I wonder if you might want to mangle that symbol name somehow. Maybe the [] in the name effectively does that, but I would really mangle it a lot like 

    format("___symname___%s[%d]", sym.mangled(), a)

Also note I use sym.mangled() instead of sym.name() so that printfs of string arrays with the same name but different scopes don't get screwed up with clashing names.

Reply:
The goal was mangling and not having variables stomp on prior. Though I think it may be entirley unnecessary as `getOrAllocateCUDAVariable` mangles string's in a unique way.